### PR TITLE
Change Project structure to be a little more scalable

### PR DESCRIPTION
**Pull request by Hennzau**
# Objective  
Hi! This is the final useful PR for #4. After ensuring everything was easy to use and to create new datatypes, I slightly changed the entire structure of the project to make it a bit more scalable. Here’s how it looks: 😎

# Usage  
To create a new datatype, you need to add a module in the `libraries/datatypes` crate. If you want to add `ndarray` or `arrow` support for this datatype, you’ll need to create a `ndarray.rs` or `arrow.rs` submodule that will use the functions and structures from the `libraries/converter` crate. For example, for **BBox**’s `arrow` support, we use the `fastformat_converter::arrow` module:

```rust
use fastformat_converter::arrow::{FastFormatArrowBuilder, FastFormatArrowRawData};

impl<'a> BBox<'a> {
    pub fn raw_data(array_data: arrow::array::ArrayData) -> Result<FastFormatArrowRawData> {
        use arrow::datatypes::Float32Type;

        let raw_data = FastFormatArrowRawData::new(array_data)?
            .load_primitive::<Float32Type>("data")?
            .load_primitive::<Float32Type>("confidence")?
            .load_utf("label")?
            .load_utf("encoding")?;

        Ok(raw_data)
    }

    pub fn from_raw_data(mut raw_data: FastFormatArrowRawData) -> Result<Self> {
        use arrow::datatypes::Float32Type;

        let data = raw_data.primitive_array::<Float32Type>("data")?;
        let confidence = raw_data.primitive_array::<Float32Type>("confidence")?;
        let label = raw_data.utf8_array("label")?;
        let encoding = Encoding::from_string(raw_data.utf8_singleton("encoding")?)?;

        Ok(Self {
            data: Cow::Owned(data),
            confidence: Cow::Owned(confidence),
            label,
            encoding,
        })
    }
}
```

We do the same for `ndarray` support. 🤓

## Features  
I’ve added the [[features]] tag for the fastformat crates, so that we don’t add a bunch of dependencies for people who don’t need them. 🚀

```toml
[dependencies]
fastformat = { version = "0.1.0", features = ["arrow", "ndarray"] }
```

### CODE REVIEW

**Comment by haixuanTao** [comment]
Relevant to this PR: https://github.com/kornia/kornia-rs/pull/115

I talked with @edgarriba and he is happy to work with us to make kornia a thing in dora-rs. And as performance is a key criteria for dora, it's interesting that if i recall correctly kornia can be slightly faster than opencv.

File: libraries/converter/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#[cfg(feature = "arrow")]
+pub mod arrow;
+
+#[cfg(feature = "ndarray")]
+pub mod ndarray;
```
Comment (edgarriba) [inline libraries/converter/src/lib.rs:5]: would be great to support a kornia Image/Tensor converter 
- image: https://github.com/kornia/kornia-rs/tree/main/crates/kornia-image/src
- tensor: https://github.com/kornia/kornia-rs/tree/main/crates/kornia-image/src

File: examples/dummy-opencv-capture/src/main.rs
Comment (edgarriba) [inline examples/dummy-opencv-capture/src/main.rs:1]: you might want to give a shot to the kornia camera streamer:

- rtsp: https://github.com/kornia/kornia-rs/tree/main/examples/rtspcam
- v4l2: https://github.com/kornia/kornia-rs/tree/main/examples/webcam

File: libraries/datatypes/src/bbox.rs
Comment (edgarriba) [inline libraries/datatypes/src/bbox.rs:1]: same here, some support conversion from kornia-dnn (to landed this week).
Still iterating on the `Detection` struct, but for now what we have is sufficient enough to keep it lightweighted

- https://github.com/kornia/kornia-rs/pull/129

Comment (edgarriba) [reply libraries/datatypes/src/bbox.rs:1]: @Hennzau @haixuanTao the kornia-dnn basically is a wrapper around the [ort](https://ort.pyke.io/) crate which I'm iterating with the author in order to run inference the models from kornia-pytorch-models from the HF hub: https://huggingface.co/kornia/ONNX_models that @shijianjian has been working on

**Comment by Hennzau** [comment]
Adding Kornia is a really good idea, I made this PR so it's easy to add/change conversion wrapper!

I opened #8 to make all the needed changes :)

**Pull request merged**